### ISSUE TITLE
Accessibility: search menu items improvements

### ISSUE THREAD

**Issue by afercia** [voldemortensen/menu-customizer#89]
When adding new links, you can search existing links in the "Search Menu Items". In its current form this search is barely usable for screen reader users. A couple of things could improve the experience:
- the "search as you type" script should try to detect when users actually end typing and debounce the search, see for example what was done in core for the Themes search: https://core.trac.wordpress.org/ticket/26600
- the search results (or no results) should be announced to assistive technologies, possible use case for `wp.a11y.speak` see same core ticket 26600
- a short description of the "live" search functionality could help, see the Themes search in core, i.e.:
`aria-describedby="live-search-desc"`

**Comment by designsimply** [comment]
Moved to https://core.trac.wordpress.org/ticket/32720
